src/drc/api/serializers.py

```diff
@@ -668,10 +669,11 @@ class ObjectInformatieObjectSerializer(serializers.HyperlinkedModelSerializer):
         queryset=EnkelvoudigInformatieObject.objects,
         help_text=get_help_text("datamodel.ObjectInformatieObject", "informatieobject"),
     )
+    informatieobject_versie = serializers.IntegerField(required=False, allow_null=True, default=None)
```

Comment:
This should be a read-only field, since you're effectively silently blocking writes already anyway.

However, I think [this](https://www.django-rest-framework.org/api-guide/fields/#default) mechanism should be explored to set the default value from the received URL.

I'm not sure if the URL is validated first or not, but you should be able to access `serializer_field.parent.initial_data` to get the value for the version.

I'd also consider tapping into `EnkelvoudigInformatieObjectHyperlinkedRelatedField` to resolve the correct DB object, including `?versie=` from the URL. Doing that, you move the `versie` validation to the `informatieobject` field, then:

1. you know that a valid version was specified
2. you know a valid document URL was specified
3. after that, the serializer-level validation is called, and the `attrs` include the correct EIO instance. You can then move on to perform unique-together validation, since you now have `EIO.canonical` and `EIO.versie`.
4. once all that is valid, at the end of the serializer `validate` method you can inject the version IF it was part of the `initial_data.informatieobject` URL

    ```py
    def validate(self, attrs):
        ...
        if "versie" in query_params:
            attrs["informatieobject_versie"] = attrs["informatieobject"].versie
        return attrs
    ```

Reply:
As mentioned on slack: I couldn't find a nice way of making EnkelvoudigInformatieObjectHyperlinkedRelatedField do the filtering on URL query_params (all ways that I found were quite hacky, since HyperlinkedRelatedField  doesn't seem to handle query params for filtering, they are removed here https://github.com/encode/django-rest-framework/blob/master/rest_framework/relations.py#L350). So I left the `InformatieObjectVersionValidator`.